users/models.py

```diff
@@ -44,6 +50,17 @@ def friends_posts(self):
                           (a.posts.all() for a in fs)
                       ).order_by('-pk'))
 
+    def authors_posts(self, user):
+        """
+        Returns all of an author's posts
+        If the user is view thier own profile, all posts are returned regardless of permissions
+        If viewing another author's profile, only public posts are returned
+        """
+        if (self.pk == user.pk):
```

Comment:
this will not work if self is an author and user is a user. if `user` is a user, change it to `self.user == user`

we generally want to compare users, because some users do not have an author, and thus are equivalent to not being logged in

Reply:
Fixed